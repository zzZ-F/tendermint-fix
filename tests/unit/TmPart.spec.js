import { mount } from "@vue/test-utils"
import { TmPart } from "@/index"

describe("TmPart", () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(TmPart, { propsData: { title: "Test" } })
  })

  it("has the expected html structure", () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it("only has header if a title is defined", () => {
    wrapper = mount(TmPart, {
      slots: { default: "<div class='custom-main' />" }
    })
    expect(wrapper.find("header").exists()).toBe(false)
    expect(wrapper.find(".custom-main").exists()).toBe(true)
  })

  it("gets title from prop if defined", () => {
    wrapper = mount(TmPart, { propsData: { title: "Test" } })
    expect(wrapper.find(".tm-part-title").text()).toBe("Test")
  })

  it("should use slots", () => {
    wrapper = mount(TmPart, {
      slots: {
        title: "<div class='custom-title' />",
        default: "<div class='custom-main' />",
        menu: "<div class='custom-menu' />"
      }
    })
    expect(wrapper.find(".custom-title").exists()).toBe(true)
    expect(wrapper.find(".custom-menu").exists()).toBe(true)
    expect(wrapper.find(".custom-main").exists()).toBe(true)
  })
})
