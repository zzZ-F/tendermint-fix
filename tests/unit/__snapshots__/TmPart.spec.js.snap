// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmPart has the expected html structure 1`] = `
<section
  class="tm-part"
>
  <div
    class="tm-part-container"
  >
    <header
      class="tm-part-header"
    >
      <div
        class="tm-part-header-container"
      >
        <div
          class="tm-part-title h5"
        >
          Test
        </div>
        <menu
          class="tm-part-menu"
        />
      </div>
    </header>
    <main
      class="tm-part-main"
    />
  </div>
</section>
`;
